block/meta-iosched.c

```diff
@@ -0,0 +1,75 @@
+#include <linux/blkdev.h>
+#include <linux/elevator.h>
+#include <linux/slab.h>
+
+#include "meta-iosched.h"
+
+struct queue_arr {
+	struct request_queue **queues;
+	int n_queues;
+	int size;
+};
+
+struct queue_arr all = {0};
+struct queue_arr major = {0};
+
+bool add_queue(struct request_queue *q) {
+	struct request_queue **temp;
+
+	if (all.size > all.n_queues) {
+		all.queues[all.n_queues++] = q;
+		return true;
+	}
+
+	temp = kmalloc((all.n_queues + 1) * 2 * sizeof(q), GFP_KERNEL);
+	if (!temp) return false;
+	all.size = (all.n_queues + 1) * 2;
+	memcpy(temp, all.queues, all.n_queues * sizeof(q));
+	kfree(all.queues);
+	all.queues = temp;
+	all.queues[all.n_queues++] = q;
+	return true;
+}
+
+int find_queue(struct queue_arr arr, struct request_queue *q) {
+	int i;
+	for (i = 0; i < arr.n_queues; i++)
+		if (arr.queues[i] == q) return i;
+	return -1;
+}
+
+bool del_queue2(struct queue_arr arr, struct request_queue *q) {
+	int pos;
+	pos = find_queue(arr, q);
+	if (pos == -1) return false;
+	arr.queues[pos] = arr.queues[arr.n_queues - 1];
+	arr.n_queues--;
+	return true;
+}
+
+bool del_queue(struct request_queue *q) {
+	del_queue2(major, q);
+	return del_queue2(all, q);
+}
+
+// Here must be smth smart :-)
+void print_stats(void) {
+	int i;
+	printk("Stats! ");
```

Comment:
printk needs KERN_&lt;something&gt; (e.g. KERN_INFO) before its format string. Or you could use pr_&lt;something&gt; (e.g. pr_info).